RealmTasks Apple/RealmTasks macOS/ItemCellView.swift

```diff
@@ -0,0 +1,499 @@
+                }
+            case nil:
+                completed ? textView.strike() : textView.unstrike()
+
+                animationBlock = {
+                    self.contentView.frame.origin.x = 0
+                }
+
+                completionBlock = {}
+            }
+
+            NSView.animate(animations: animationBlock) {
+                completionBlock()
+
+                self.doneIconView.frame.origin.x = originalDoneIconOffset
+                self.deleteIconView.frame.origin.x = originalDeleteIconOffset
+
+                self.isUserInteractionEnabled = true
+            }
+        default:
+            break
+        }
+    }
+
+}
+
+// MARK: Private Classes
+
+private enum ReleaseAction {
+    case Complete, Delete
+}
+
+private class HighlightView: NSView {
+
+    private override func drawRect(dirtyRect: NSRect) {
+        let shadowColor = NSColor(white: 0, alpha: 0.2)
+        let backgroundColor = NSColor(white: 0, alpha: 0.05)
+
+        let gradient = NSGradient(colorsAndLocations: (shadowColor, 0), (backgroundColor, 0.08), (backgroundColor, 0.92), (shadowColor, 1))
+        gradient?.drawInRect(bounds, angle: 90)
+    }
+
+}
+
+protocol ItemTextFieldDelegate: NSTextFieldDelegate {
+
+    func textFieldDidBecomeFirstResponder(textField: NSTextField)
+
+}
+
+private class ItemTextField: NSTextField {
+
+    override init(frame frameRect: NSRect) {
+        super.init(frame: frameRect)
+
+        bordered = false
+        focusRingType = .None
+        font = .systemFontOfSize(18)
+        textColor = .whiteColor()
+        backgroundColor = .clearColor()
+        lineBreakMode = .ByWordWrapping
+        selectable = false
+        editable = false
+    }
+
+    required init?(coder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+
+    private override func becomeFirstResponder() -> Bool {
+        (delegate as? ItemTextFieldDelegate)?.textFieldDidBecomeFirstResponder(self)
+
+        let result = super.becomeFirstResponder()
+
+        currentEditor()?.selectedRange = NSRange(location: stringValue.characters.count, length: 0)
+
+        return result
+    }
+
+    override var intrinsicContentSize: NSSize {
+        // By default editable NSTextField doesn't respect wrapping while calculating intrinsic content size,
+        // let's calculate the correct one by ourselves
+        let placeholderFrame = NSRect(origin: .zero, size: NSSize(width: frame.width, height: .max))
+        let calculatedHeight = cell!.cellSizeForBounds(placeholderFrame).height
+
+        return NSSize(width: frame.width, height: calculatedHeight)
+    }
+
+    override func textDidChange(notification: NSNotification) {
+        super.textDidChange(notification)
+
+        // Update height on text change
+        invalidateIntrinsicContentSize()
+    }
+
+}
+
+// MARK: Private Extensions
+
+private extension NSTextField {
+
+    func strike(fraction: Double = 1) {
```

Comment:
These functions could be mostly shared across iOS and macOS if they were on `NSAttributedString` instead of `NSTextField`/`CellTextView`.